File: lib/src/main/java/io/github/kory33/guardedqueries/core/subqueryentailments/LocalInstanceTermFact.java
package io.github.kory33.guardedqueries.core.subqueryentailments;

import io.github.kory33.guardedqueries.core.formalinstance.FormalFact;
import uk.ac.ox.cs.pdq.fol.Atom;
import uk.ac.ox.cs.pdq.fol.Variable;

import java.util.function.Function;

public class LocalInstanceTermFact {
    private LocalInstanceTermFact() {
    }

    public static FormalFact<LocalInstanceTerm> fromAtomWithVariableMap(
            final Atom fact,
            final Function<? super Variable, ? extends LocalInstanceTerm> mapper
    ) {
        return FormalFact.fromAtom(fact).map(term -> LocalInstanceTerm.fromTermWithVariableMap(term, mapper));
    }
}

File: lib/src/main/scala/io/github/kory33/guardedqueries/core/subqueryentailments/LocalInstanceTermFact.scala
package io.github.kory33.guardedqueries.core.subqueryentailments

import io.github.kory33.guardedqueries.core.formalinstance.FormalFact
import uk.ac.ox.cs.pdq.fol.Atom
import uk.ac.ox.cs.pdq.fol.Variable

object LocalInstanceTermFact {
  def fromAtomWithVariableMap(
    fact: Atom,
    mapper: Variable => LocalInstanceTerm
  ): FormalFact[LocalInstanceTerm] = FormalFact.fromAtom(fact).map(term =>
    LocalInstanceTerm.fromTermWithVariableMap(term, mapper)
  )
}
